src/main/java/rx/observers/TestSubscriber.java

```diff
@@ -670,4 +673,33 @@ final void assertionError(String message) {
         }
         throw ae;
     }
+    
+    /**
+     * Assert that the TestSubscriber contains the given first and optional rest values exactly
+     * and if so, clears the internal list of values.
+     * <p>
+     * <code><pre>
+     * TestSubscriber ts = new TestSubscriber();
+     * 
+     * ts.onNext(1);
+     * 
+     * ts.assertAndConsume(1);
+     * 
+     * ts.onNext(2);
+     * ts.onNext(3);
+     * 
+     * ts.assertAndConsume(2, 3); // no mention of 1
+     * </pre></code>
+     * @param expectedFirstValue the expected first value
+     * @param expectedRestValues the optional rest values
+     */
+    public final void assertAndConsume(T expectedFirstValue, T... expectedRestValues) {
```

Comment:
That implies the whole internal state is wiped out and you want to reuse the same instance on another Observable.